### ISSUE TITLE
L1 distance?

### ISSUE THREAD

**Issue by MBkkt** [ashvardanian/SimSIMD#133]
Manhattan distance sometimes useful

https://github.com/google-research/google-research/tree/master/scann/scann/distance_measures/one_to_one
https://github.com/ydb-platform/ydb/tree/main/library/cpp/l1_distance

**Comment by ashvardanian** [comment]
This might be interesting for `i8` inputs and any other low-capacity integral types, and might be related to #100

**Comment by raulod** [comment]
l1 is picking momentum is many high dimensional usage where l2 distances are unstable.

**Comment by ashvardanian** [comment]
It seems like L1 is still very niche and L2 & Angular distances are mostly used.

Would be great to see some more user requests before switching the attention from matrix multiplications.

**Comment by rvalavi** [comment]
Is there any chance that L1 distance will be added for f32 or bf16?

**Comment by ashvardanian** [comment]
Yes, but depends on the demand. Where do you plan to use it, @rvalavi?

**Comment by rvalavi** [comment]
Thanks for the quick response, @ashvardanian!

L1 distance is pretty common in ecology and environmental modeling, especially when you’re working with multidimensional distances. E.g.: https://github.com/fitzLab-AL/gdm

I’m in the process of writing a Rust crate for this kind of application (private for now, but planning to make it public soon), and I wanted to try this out. In another application I worked on, I couldn’t get Rust anywhere close to the performance of a C++ implementation using Eigen with SIMD enabled, which is why having support for things like L1 here matters to me.

It’d be great to have L1 distance supported, since it comes up a lot in real-world use cases like this. I really appreciate you taking a look.
